reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -0,0 +1,677 @@
+
+        publisher.assertWasRequested();
+        publisher.assertNoRequestOverflow();
+
+        Assert.assertEquals(2L, requested.get());
+    }
+
+    @Test
+    public void backpressureConditionalTest() {
+        Flux<Integer> publisher = Flux.range(0, 10000);
+        AtomicLong requested = new AtomicLong();
+
+        Flux<String> switchTransformed = publisher.doOnRequest(requested::addAndGet)
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf))
+                                                  .filter(e -> false);
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+
+        Assert.assertEquals(2L, requested.get());
+
+    }
+
+    @Test
+    public void backpressureHiddenConditionalTest() {
+        Flux<Integer> publisher = Flux.range(0, 10000);
+        AtomicLong requested = new AtomicLong();
+
+        Flux<String> switchTransformed = publisher.doOnRequest(requested::addAndGet)
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf)
+                                                                                                .hide())
+                                                  .filter(e -> false);
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+
+        Assert.assertEquals(10001L, requested.get());
+    }
+
+    @Test
+    public void backpressureDrawbackOnConditionalInTransformTest() {
+        Flux<Integer> publisher = Flux.range(0, 10000);
+        AtomicLong requested = new AtomicLong();
+
+        Flux<String> switchTransformed = publisher.doOnRequest(requested::addAndGet)
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux
+                                                          .map(String::valueOf)
+                                                          .filter(e -> false));
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectComplete()
+                    .verify(Duration.ofSeconds(10));
+
+        Assert.assertEquals(10001L, requested.get());
+    }
+
+    @Test
+    public void shouldErrorOnOverflowTest() {
+        TestPublisher<Long> publisher = TestPublisher.createCold();
+
+        Flux<String> switchTransformed = publisher.flux()
+                                                  .switchOnFirst((first, innerFlux) -> innerFlux.map(String::valueOf));
+
+        publisher.next(1L);
+
+        StepVerifier.create(switchTransformed, 0)
+                    .thenRequest(1)
+                    .expectNext("1")
+                    .then(() -> publisher.next(2L))
+                    .expectError()
```

Comment:
maybe assert the error more deeply (class, message)?

Reply:
done